### PR TITLE
BRS-729: Error searching by reservation number

### PR DESCRIPTION
**Pull request by molund**
### Jira Ticket:

BRS-729

### Jira Ticket URL:

https://bcparksdigital.atlassian.net/browse/BRS-729

### Description:

Based on my testing, searching by reservation number never worked (it didn’t work sometimes as stated in the ticket title)  

The actual error was **“Conditions can be of length 1 or 2 only”**

This query was causing the error:

```
{
   "TableName":"parksreso",
   "ExpressionAttributeValues":{
      ":shortPassDate":{
         "S":"2022-07-21"
      },
      ":facilityName":{
         "S":"Test"
      },
      ":passType":{
         "S":"DAY"
      },
      ":sk":{
         "S":"055291844"
      }
   },
   "IndexName":"shortPassDate-index",
   "KeyConditionExpression":"shortPassDate =:shortPassDate AND facilityName =:facilityName AND sk =:sk",
   "FilterExpression":"#theType =:passType",
   "ExpressionAttributeNames":{
      "#theType":"type"
   }
}
```

I moved the ` AND sk =:sk` check from `KeyConditionExpression` to `FilterExpression` to and changed it to check for ` AND #registrationNumber =:registrationNumber` instead.  This matches the pattern being used by firstName, lastName, email, etc.

```
{
   "TableName":"parksreso",
   "ExpressionAttributeValues":{
      ":shortPassDate":{
         "S":"2022-07-21"
      },
      ":facilityName":{
         "S":"Test"
      },
      ":passType":{
         "S":"DAY"
      },
      ":registrationNumber":{
         "S":"055291844"
      }
   },
   "IndexName":"shortPassDate-index",
   "KeyConditionExpression":"shortPassDate =:shortPassDate AND facilityName =:facilityName",
   "FilterExpression":"#theType =:passType AND #registrationNumber =:registrationNumber",
   "ExpressionAttributeNames":{
      "#theType":"type",
      "#registrationNumber":"registrationNumber"
   }
}
```

### CODE REVIEW

**Review by cameronpettit — APPROVED** [review]
Tested this after rebasing the BRS-673 changes and is working as expected. Will need to be rebased before merging, but we should probably hold out on that until the 673 changes are fully approved.

**Review by meredithom — COMMENTED** [review]
Looks like it's working as intended before the 673 and accompanying changes. I will approve this once all of the 673 changes have been approved. 

As a future feature, it might be nice to include the ability to search for part of the number instead of requiring the full thing.

File: lambda/readPass/index.js
Comment (marklise) [inline lambda/readPass/index.js:87]: This is not needed, registrationNumber is not a reserved word.

Comment (molund) [reply lambda/readPass/index.js]: This has been updated so it no longer uses ExpressionAttributeNames.  
searchLastName, searchFirstName, and email were also updated since they also are not reserved words.

File: lambda/readPass/index.js
Comment (marklise) [inline lambda/readPass/index.js:93]: `registrationNumber =:reservationNumber` shouldn't this be `registrationNumber =:registrationNumber `

Comment (molund) [reply lambda/readPass/index.js:93]: The column is named incorrectly in the database. It's really a `reservationNumber` not a `registrationNumber`.  There should probably be a tech-debt ticket created to rename it later and migrate the data.

Comment (marklise) [reply lambda/readPass/index.js:93]: Good idea.

Comment (molund) [reply lambda/readPass/index.js:93]: Either way it's inconsistent.  It just depends if we want the inconsistency on lines 87+88, or on line 93.  But I can move the inconsistency back to 87+88 since that's where it was originally. 
I have created a ticket BRS-748 to address the tech-debt.

Comment (marklise) [reply lambda/readPass/index.js]: Yeah originally I was just highlighting the #reservedWord change, so I was surprised to see name change come in afterward. I'd prefer to leave it as it was before the nit and address it in the ticket you just created.

**Review by marklise — APPROVED** [review]

**Pull request merged**
